includes/wikia/api/DWDimensionApiController.class.php

```diff
@@ -225,13 +227,53 @@ public function getWikiCategories() {
 	}
 
 	public function getWikiImages() {
-		$this->getDataPerWiki( array( $this, 'getWikiImagesData' ) );
+
+		$startingWikiId = $this->getSharedDbSlave()->strencode( $this->getRequest()->getVal( 'starting_wiki_id',
+			static::DEFAULT_AFTER_ID ) );
+		$afterImageName = $this->getSharedDbSlave()->strencode( $this->getRequest()->getVal(
+			'after_image_name', static::DEFAULT_AFTER_IMAGE_NAME ) );
+		$limit = min( $this->getSharedDbSlave()->strencode( $this->getRequest()->getVal(
+			'limit', static::LIMIT ) ), static::LIMIT_MAX );
+
+		$wikis = $this->getWikiDbNames( [ "city_id >= ".$startingWikiId ] );
+
+		$result = [];
+		foreach( $wikis as $wiki ) {
+			$db = $this->getWikiConnection( $wiki[ 'cluster' ], $wiki[ 'dbname' ] );
+			$sub_result = null;
+			if ( isset( $db ) ) {
+				$sub_result = $this->getWikiImagesData( $db, $afterImageName, $limit );
+			}
+			$result[] = [
+				'wiki_id' => $wiki[ 'wiki_id' ],
+				'data' => $sub_result
+			];
+			$afterImageName = '';
+			if ( $limit > 0 ) {
+				$limit = $limit - count( $sub_result );
+				if ( $limit <= 0 ) {
+					break;
+				}
+			}
+		}
+		foreach( $this->connections as $connection ) {
+			$connection->close();
+		}
+
+		$this->setResponseData(
+			$result,
+			null,
+			WikiaResponse::CACHE_DISABLED
+		);
 	}
 
-	private function getWikiImagesData( DatabaseMysqli $db ) {
+	private function getWikiImagesData( DatabaseMysqli $db, $afterImageName, $limit ) {
 		$result = [];
 		try {
-			$rows = $db->query( DWDimensionApiControllerSQL::DIMENSION_WIKI_IMAGES, __METHOD__ );
+			$query = DWDimensionApiControllerSQL::DIMENSION_WIKI_IMAGES;
+			$query = str_replace( '$limit', $limit, $query);
+			$query = str_replace( '$img_name', $afterImageName, $query);
```

Comment:
same here

Reply:
we have intellij code style config that might help: https://github.com/Wikia/guidelines/blob/master/PHP/bin/MediaWiki.xml